appsec/src/extension/tags.c

```diff
@@ -960,6 +970,12 @@ static zval *nullable _root_span_get_meta(void)
     return meta;
 }
 
+static void _emit_auto_user_event(void)
+{
+    dd_tags_set_sampling_priority();
+    dd_trace_emit_asm_event();
```

Comment:
Isn't this duplicative? `dd_tags_set_sampling_priority` sets `force_keep` so that later we can call:

```c
dd_trace_set_priority_sampling_on_span_zobj(
    span, PRIORITY_SAMPLING_USER_KEEP, DD_MECHANISM_ASM);
```

But by calling `dd_trace_emit_asm_event()` we already do:

```c
if (DDTRACE_G(active_stack)->root_span) {
    DDTRACE_G(active_stack)->root_span->asm_event_emitted = true;
    ddtrace_trace_source_set_asm_source();
} else {
    DDTRACE_G(asm_event_emitted) = true;
}
 ddtrace_set_priority_sampling_on_root(PRIORITY_SAMPLING_USER_KEEP, DD_MECHANISM_ASM);
```